.buildbot.sh

```diff
@@ -33,5 +21,27 @@ sh rustup.sh --default-host x86_64-unknown-linux-gnu \
     --profile minimal \
     -y
 export PATH=${CARGO_HOME}/bin/:$PATH
+
+# Build the compiler and add it as a linked toolchain.
+git clone ${YKRUSTC_REPO}
+cd ykrustc
+cp .buildbot.config.toml config.toml
+./x.py build --stage 1
+rustup toolchain link ykrustc-stage1 `pwd`/build/x86_64-unknown-linux-gnu/stage1
+cd ..
+rustup override set ykrustc-stage1
+
+# Test both workspaces using the compiler we just built.
+export RUSTFLAGS="-C tracer=${CI_TRACER_KIND} -D warnings"
+cargo xtask test
+cargo xtask bench
+cargo xtask clean
+
+# Also test the build without xtask, as that's what consumers will do.
+cargo build
```

Comment:
Does this work -- I thought we *had* to use `xtask`?

Reply:
Yes it does work for `cargo build`, but all warnings and build progress for the internal workspace is hidden as build script output is only shown when it fails.

Reply:
We tested the xtask case a few lines above. This is testing what crates consuming us will do (we have no way to have consumers run xtask, so we had to duplicate some code in `build.rs` -- it's complicated).